internal/models/helpers/strlistattr/strlist.go

```diff
@@ -0,0 +1,49 @@
+package strlistattr
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listdefault"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func Optional(validators ...validator.List) schema.ListAttribute {
+	return schema.ListAttribute{
+		Optional:    true,
+		Computed:    true,
+		ElementType: types.StringType,
+		Default:     listdefault.StaticValue(types.ListNull(types.StringType)),
+		Validators:  validators,
+	}
+}
+
+func Get(s []string, data map[string]any, key string) {
+	data[key] = s
+}
+
+func Set(s *[]string, data map[string]any, key string) {
+	if v, ok := data[key].([]any); ok {
+		if len(v) > 0 {
+			for i := range v {
+				str, ok := v[i].(string)
+				if !ok {
+					panic(fmt.Sprintf("unexpected value of type %T in string list: %s", v[i], key))
```

Comment:
this is how we handle errors in other setters as well?
theoretically there could be non string values there, if developer interferes with the state file? not sure

Reply:
The `data` is JSON received from the Descope backend, so if there's a type mismatch here it's a programmer error on our end. Moreover it should be caught in acceptance tests.